### PR TITLE
chore: enhance Source Detection Logic in Analytics Parameters

### PR DESCRIPTION
**Pull request by omridan159**
## **Description**

Refactored the logic for determining the source in the analytics parameters in the `AccountApproval` component.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/AccountApproval/index.js [hunk 1]
```diff
@@ -121,6 +121,30 @@ class AccountApproval extends PureComponent {
       console.error('URL conversion error:', error);
     }
 
+    const getSource = () => {
```
Comment (andreahaku) [inline app/components/UI/AccountApproval/index.js:124]: to simplify:
```
const getSource = (): string => {
  const source = currentPageInformation?.analytics?.source;
  
  if (source !== undefined) {
    return source;
  }

  if ('source' in currentPageInformation?.analytics && source === undefined) {
    return 'dappDeepLinkUrl';
  }

  return this.props.walletConnectRequest ? 'walletconnect' : 'sdk';
};
```

**Review by andreahaku — COMMENTED** [review]
simplified the `getSource` function

File: app/components/UI/AccountApproval/index.js [hunk 2]
```diff
@@ -121,6 +122,32 @@ class AccountApproval extends PureComponent {
...
+        currentPageInformation.analytics.source === undefined;
+
+      const hasSource = currentPageInformation?.analytics?.source !== undefined;
+
+      if (hasSource) {
+        return currentPageInformation.analytics.source;
+      }
+
+      if (hasNoSource) {
+        return 'dappDeepLinkUrl';
```
Comment (andreahaku) [inline app/components/UI/AccountApproval/index.js:138]: add it to SourceType

**Review by andreahaku — COMMENTED** [review]
missing field in SourceTyoe

File: app/components/UI/AccountApproval/index.js (hunk 1 above)
Comment (omridan159) [reply app/components/UI/AccountApproval/index.js:125]: it will throw error

Comment (omridan159) [reply app/components/UI/AccountApproval/index.js:125]: Here => 'source' in currentPageInformation?.analytics

if analytics === undefined

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**